src/helpers.ts

```diff
@@ -64,7 +65,7 @@ export function merge<T>(target: T, src: T, shallow?: boolean): T {
                 const targetProp = target[k];
                 const srcProp = src[k];
                 if (Array.isArray(targetProp) && Array.isArray(srcProp)) {
-                    mergeNamedArrays(targetProp, srcProp);
+                    mergeNamedArrays(targetProp as any, srcProp as any);
```

Comment:
Could be better to wait for https://github.com/microsoft/TypeScript/issues/40463 because the fix will probably be done earlier than the merge here 😁

Reply:
Oh I'm wrong, the merges just happened!